### PR TITLE
bpo-45565: Specialize LOAD_ATTR_CLASS

### PR DESCRIPTION
**Pull request by Fidget-Spinner**
Almost same as `LOAD_METHOD_CLASS`.

https://bugs.python.org/issue45565

### CODE REVIEW

File: Python/ceval.c [hunk 1]
```diff
@@ -1477,6 +1477,21 @@ eval_frame_handle_pending(PyThreadState *tstate)
     STAT_INC(LOAD_##attr_or_method, hit); \
     Py_INCREF(res);
 
+// shared by LOAD_ATTR_CLASS and LOAD_METHOD_CLASS
+#define LOAD_CLASS_ATTR_OR_METHOD(attr_or_method) \
```
Comment (markshannon) [inline Python/ceval.c]: Maybe drop this macro, it makes it a bit harder to understand `LOAD_ATTR_CLASS` and `LOAD_METHOD_CLASS` as you need to jump back and forth when reading them.

If you are worried about code length, the code could be made a bit shorter, by grouping the asserts and dropping the one-use temporaries:
```C
    SpecializedCacheEntry *caches = GET_CACHE();
    DEOPT_IF(!PyType_Check(cls), LOAD_ATTR);
    DEOPT_IF(((PyTypeObject *)cls)->tp_version_tag != caches[0].attr.version,  LOAD_ATTR);
    ASSERT(valid_class_attr_cache(caches));
    STAT_INC(LOAD_ATTR, hit);
    PyObject *res = caches[-1].obj.obj;
    Py_INCREF(res);
```

Comment (Fidget-Spinner) [reply Python/ceval.c]: I went with approach 2 that doesn't require this anymore.

**Comment by iritkatriel** [comment]
@Fidget-Spinner Is this abandoned, or are you planning to continue working on it?

**Comment by Fidget-Spinner** [comment]
@iritkatriel this was completed in https://github.com/python/cpython/pull/93430). Thanks for the reminder!

**Pull request closed without merging**